internal/server/server.go

```diff
@@ -210,7 +210,12 @@ func (h *Handlers) ComposeImage(ctx echo.Context) error {
 
 	repositories, err := RepositoriesForImage(h.server.distsDir, composeRequest.Distribution, composeRequest.ImageRequests[0].Architecture)
 	if err != nil {
-		return err
+		switch err.Error() {
+		case "Architecture not supported":
```

Comment:
I don't think we should be relying on the error message in order to return the correct error. 

Imo there's 2 options:
1. Since 'distribution.go' is still in `internal/server`, how about we just call `echo.NewHTTPError` inside the distribution?

2. Alternatively we define our own set of errors which can then be compared compliant with https://github.com/golang/go/wiki/ErrorValueFAQ (Error.Is())

I think I prefer 1 for simplicity's sake. Since we'd just end up redefining all possible http errors.

Reply:
I agree with you, I think 1 is preferable - thanks for the input!